# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/power/WakeLockJS.cpp
# Commit: a0db8be67659
# Full Hash: a0db8be67659d142275051d77ba9829dca6cf521
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1589554
# File Overlap Count: 1
# Description:
#   Bug 1589554 - Part 7: Add logging to Screen Wake Lock. r=dom-core,edgar
#   
#   Depends on D189513
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189514
# ==============================================================================

diff -r 76a3c248813f -r a0db8be67659 dom/power/WakeLockJS.cpp
--- a/dom/power/WakeLockJS.cpp	Tue Dec 05 15:37:07 2023 +0000
+++ b/dom/power/WakeLockJS.cpp	Tue Dec 05 15:37:07 2023 +0000
@@ -7,6 +7,7 @@
 #include "ErrorList.h"
 #include "mozilla/AlreadyAddRefed.h"
 #include "mozilla/Assertions.h"
+#include "mozilla/Logging.h"
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/Event.h"
 #include "mozilla/dom/EventTarget.h"
@@ -30,6 +31,8 @@
 
 namespace mozilla::dom {
 
+static mozilla::LazyLogModule sLogger("ScreenWakeLock");
+
 #define MIN_BATTERY_LEVEL 0.05
 
 nsLiteralCString WakeLockJS::GetRequestErrorMessage(RequestError aRv) {
@@ -103,6 +106,7 @@
   RefPtr<WakeLockSentinel> kungFuDeathGrip = aLock;
   aDoc->ActiveWakeLocks(aType).Remove(aLock);
   aLock->NotifyLockReleased();
+  MOZ_LOG(sLogger, LogLevel::Debug, ("Released wake lock sentinel"));
 }
 
 NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE(WakeLockJS, mWindow)
@@ -159,6 +163,7 @@
 // https://w3c.github.io/screen-wake-lock/#the-request-method
 already_AddRefed<Promise> WakeLockJS::Request(WakeLockType aType,
                                               ErrorResult& aRv) {
+  MOZ_LOG(sLogger, LogLevel::Debug, ("Received request for wake lock"));
   nsCOMPtr<nsIGlobalObject> global = mWindow->AsGlobal();
 
   RefPtr<Promise> promise = Promise::Create(global, aRv);
@@ -180,6 +185,8 @@
         if (lockOrErr.isOk()) {
           RefPtr<WakeLockSentinel> lock = lockOrErr.unwrap();
           promise->MaybeResolve(lock);
+          MOZ_LOG(sLogger, LogLevel::Debug,
+                  ("Resolved promise with wake lock sentinel"));
         } else {
           promise->MaybeRejectWithNotAllowedError(
               GetRequestErrorMessage(lockOrErr.unwrapErr()));
